pytests/test_execution.py

```diff
@@ -46,73 +42,52 @@ def boom(item):
     assert len(out) < 3
 
 
-@mark.skipif(
-    os.name == "nt",
-    reason=(
-        "Sending os.kill(test_proc.pid, signal.CTRL_C_EVENT) sends event to all"
-        " processes on this console so interrupts pytest itself"
-    ),
-)
-def test_can_be_ctrl_c(mp_ctx, entry_point):
-    with mp_ctx.Manager() as man:
-        is_running = man.Event()
-        out = man.list()
-
-        def proc_main():
-            flow = Dataflow()
-            inp = range(1000)
-            flow.input("inp", TestingInput(inp))
-
-            def mapper(item):
-                is_running.set()
-                return item
-
-            flow.map(mapper)
-            flow.output("out", TestingOutput(out))
-
-            try:
-                entry_point(flow)
-            except KeyboardInterrupt:
-                exit(99)
-
-        test_proc = mp_ctx.Process(target=proc_main)
-        test_proc.start()
-
-        assert is_running.wait(timeout=5.0), "Timeout waiting for test proc to start"
-        os.kill(test_proc.pid, signal.SIGINT)
-        test_proc.join(timeout=10.0)
-
-        assert test_proc.exitcode == 99
-        assert len(out) < 1000
-
-
-def test_requires_input(entry_point):
-    flow = Dataflow()
-    out = []
-    flow.output("out", TestingOutput(out))
-
-    with raises(ValueError):
-        entry_point(flow)
-
-
-def test_requires_output(entry_point):
-    flow = Dataflow()
-    inp = range(3)
-    flow.input("inp", TestingInput(inp))
-
-    with raises(ValueError):
-        entry_point(flow)
-
-
-@mark.skip(
-    "We should have a way to access worker_index in whatever replaces spawn_cluster"
-)
-def test_input_parts_hashed_to_workers(mp_ctx):
-    pass
-
-
-@mark.skip(
-    "We should have a way to access worker_index in whatever replaces spawn_cluster"
-)
-def test_output_parts_hashed_to_workers(mp_ctx):
-    pass
+def test_cluster_can_be_ctrl_c():
```

Comment:
I think it makes sense to test `spawn_cluster` here directly. And then have some separate tests for the CLI wrapper function that's a little more light weight.

Reply:
There is a hidden detail that breaks this: since we need to respawn a new process with the same parameters multiple times, `spawn_cluster` under the hood calls the process with the same command line arguments. During a pytest run this means that spawn_cluster calls pytest again rather than "python -m bytewax.run", so if we want to test the script we need to test it "from outside". `spawn_cluster` should not be used directly outside of `bytewax.run`.

Another approach would be to force running "python -m bytewax.run" and manually rebuild the rest of the args  instead of reusing `sys.args` but this means we make some assumptions which might not always be true, like "python" being the correct executable. I'll look at this after the rest of the fixes